flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -458,7 +468,12 @@ public IcebergSource<T> build() {
       checkRequired();
       // Since builder already load the table, pass it to the source to avoid double loading
       return new IcebergSource<T>(
-          tableLoader, context, readerFunction, splitAssignerFactory, table);
+          tableLoader,
+          context,
+          readerFunction,
+          splitAssignerFactory,
+          flinkReadConf.planRetryNum(),
```

Comment:
We use `ScanContext` to pass parameters, but this breaks it.

Reply:
Thanks for the review @hililiwei!

What I have seen is that all of the `ScanContext` configurations are passed down to the Iceberg scan. This one is a Flink only configuration and we do not have to pass that down there. With that in mind, do you still think that `planRetryNum` belongs there? No strong opinions here, just feels a bit strange.

Reply:
But this is not entirely true. For example, `monitorInterval`, which is used to configure the monitoring interval, and `exposeLocality`, `planParallelism`, which are only for Flink.

Reply:
By the way, I once thought about deleting `ScanContext`, because we can use `flinkReadConf` to uniformly parse these parameters, but it was not achieved due to serialization issues. Maybe we can reconsider simplifying this part of the logic, removing `ScanContext`. Of course, this is not what this PR needs to pay attention to.

Reply:
Thank @hililiwei! Makes sense. Added the config to the `ScanContext`